tests/MongoDB.Analyzer.Tests.Common/DataModel/Fruit.cs

```diff
@@ -96,5 +96,17 @@ public class RedApple : Apple
 
         public string RedAppleType { get; set; }
     }
+
+    [BsonSerializerAttribute(typeof(Person))]
+    public class GrannyApple : Apple
+    {
+        public double GrannyAppleCost { get; set; }
+    }
+
+    [BsonKnownTypes(typeof(int))]
+    public class GoldenApple : Apple
+    {
+        public double GoldenAppleCost { get; set; }
+    }
```

Comment:
Lets add tests with attributes that reference types that can't be processed by `TypeProcessor`. So we test the "null" attributes paths you added.
 For example `[BsonId(IdGenerator = typeof(System.Type))]`

Reply:
The example "[BsonId(IdGenerator = typeof(System.Type))]" actually produces an internal exception(i.e. not a Bson/Driver exception). When debugging, the analyzer was able to reproduce the internal exception. However, since it is not a driver/bson exception and the settings for displaying internal exceptions are currently false, no diagnostic is rendered and the exception is not propagated back to the user. This is not due to a flaw in our attribute handling logic but more so due to the pre-existing limits on what exceptions are propagated back to user. 

Hence, I used some other examples that demonstrate the same kind of behavior we are aiming to test. 

Examples: 
- [BsonKnownTypes(typeof(System.Type))]
- [BsonKnownTypes(typeof(System.TimeSpan))]